cuda_bindings/cuda/bindings/_lib/windll.pxd

```diff
@@ -0,0 +1,36 @@
+# SPDX-FileCopyrightText: Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-NVIDIA-SOFTWARE-LICENSE
+
+from libc.stddef cimport wchar_t
+from libc.stdint cimport uintptr_t
+from cpython cimport PyUnicode_AsWideCharString
+
+cdef extern from "windows.h":
+    ctypedef void* HMODULE
+    ctypedef void* HANDLE
+    ctypedef void* FARPROC
+    ctypedef unsigned long DWORD
+    ctypedef const wchar_t *LPCWSTR
+
+    cdef DWORD LOAD_LIBRARY_SEARCH_SYSTEM32 = 0x00000800
+
+    HMODULE _LoadLibraryExW "LoadLibraryExW"(
+        LPCWSTR lpLibFileName,
+        HANDLE hFile,
+        DWORD dwFlags
+    ) nogil
+
+    FARPROC _GetProcAddress "GetProcAddress"(HMODULE hModule, const char* lpProcName) nogil
+
+cdef inline uintptr_t LoadLibraryExW(str path, HANDLE hFile, DWORD dwFlags) nogil:
+    cdef wchar_t* wpath
+    with gil:
+        wpath = PyUnicode_AsWideCharString(path, NULL)
```

Comment:
Sorry I did not read the docs for `PyUnicode_AsWideCharString` yesterday, which says
> Returns a buffer allocated by [PyMem_New](https://docs.python.org/3/c-api/memory.html#c.PyMem_New) (use [PyMem_Free()](https://docs.python.org/3/c-api/memory.html#c.PyMem_Free) to free it) on success.

so wouldn't this be leaking memory if we return without free? Also, we should check first if `wpath` is NULL before using it.